src/Microsoft.AspNetCore.Blazor/RenderTree/RenderTreeDiffBuilder.cs

```diff
@@ -597,9 +597,15 @@ private static void InitializeNewComponentFrame(ref DiffContext diffContext, int
 
         private static void InitializeNewAttributeFrame(ref DiffContext diffContext, ref RenderTreeFrame newFrame)
         {
-            if (newFrame.AttributeValue is UIEventHandler)
+            if (newFrame.AttributeValue is MulticastDelegate @delegate &&
+                @delegate.Method != null)
             {
-                diffContext.Renderer.AssignEventHandlerId(ref newFrame);
+                var parameters = @delegate.Method.GetParameters();
```

Comment:
I think `GetParameters` allocates (i.e., for the returned array). I'd strongly value not allocating here on every render, even at the cost of less fidelity over which delegate types we assign event handler IDs to.

Reply:
OK great. Sounds like we'll be able to avoid this.